File: 122-best-time-to-buy-and-sell-stock-ii/122.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0
        minPrice = math.inf
        currProfit = -math.inf

        for price in prices:
            if price < minPrice and currProfit == -math.inf:
                minPrice = price
            elif (price - minPrice) > currProfit:
                currProfit = price - minPrice
            else:
                profit += currProfit
                currProfit = -math.inf
                minPrice = price

        if currProfit != -math.inf:
            profit += currProfit

        return profit

File: 122-best-time-to-buy-and-sell-stock-ii/122.c
int maxProfit(int* prices, int pricesSize){
    int curr_min = 10000;
    int curr_max = 0;
    int profit = 0;

    for (int i = 0; i < pricesSize; ++i) {
        if (prices[i] < curr_min) {
            curr_min = curr_max = prices[i];
        } else if (prices[i] > curr_max) {
            curr_max = prices[i];
            profit += curr_max - curr_min;
            curr_min = curr_max;
        }
    }

    return profit;
}
